airflow/decorators/assets.py

```diff
@@ -0,0 +1,129 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import inspect
+import types
+from typing import TYPE_CHECKING, Any, Iterator, Mapping
+
+import attrs
+
+from airflow.assets import Asset, _validate_identifier
+from airflow.models.dag import DAG, ScheduleArg
+from airflow.operators.python import PythonOperator
+
+if TYPE_CHECKING:
+    from airflow.io.path import ObjectStoragePath
+
+
+@attrs.define(kw_only=True)
+class AssetRef:
+    """Reference to an asset."""
+
+    name: str
+
+
+class _AssetMainOperator(PythonOperator):
+    def __init__(self, *, definition_name: str, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self._definition_name = definition_name
+
+    def _iter_kwargs(self, context: Mapping[str, Any]) -> Iterator[tuple[str, Any]]:
+        for key in inspect.signature(self.python_callable).parameters:
+            if key == "self":
+                value: Any = AssetRef(name=self._definition_name)
+            elif key == "context":
+                value = context
+            else:
+                # TODO: This does not check if the upstream asset actually
```

Comment:
I think we should simply do nothing. It seems to be the current behavior of our inlets if they're not referenced

Reply:
Makes sense.